### PR TITLE
Better handling of per-command default values

### PR DESCRIPTION
**Pull request by dgageot**
Common flags can now have a default value that's different based on the command.

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: cmd/skaffold/app/cmd/run_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		t.NewTempDir().Chdir()
+		t.Override(&opts, config.SkaffoldOptions{})
+
+		cmd := NewCmdRun()
+		cmd.SilenceUsage = true
+		cmd.Execute()
+
+		t.CheckDeepEqual(false, opts.Tail)
+		t.CheckDeepEqual(false, opts.Force)
+		t.CheckDeepEqual(false, opts.Force)
```
Comment (briandealwis) [inline cmd/skaffold/app/cmd/run_test.go:37]: `opts.EnableRPC`?

File: cmd/skaffold/app/cmd/debug_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		t.NewTempDir().Chdir()
+		t.Override(&opts, config.SkaffoldOptions{})
+
+		cmd := NewCmdDebug()
+		cmd.SilenceUsage = true
+		cmd.Execute()
+
+		t.CheckDeepEqual(true, opts.Tail)
+		t.CheckDeepEqual(false, opts.Force)
+		t.CheckDeepEqual(false, opts.Force)
```
Comment (briandealwis) [inline cmd/skaffold/app/cmd/debug_test.go:37]: `opts.EnableRPC`?

File: cmd/skaffold/app/cmd/dev_test.go [hunk 3]
```diff
@@ -162,3 +162,18 @@ func TestDevConfigChange(t *testing.T) {
...
+		t.NewTempDir().Chdir()
+		t.Override(&opts, config.SkaffoldOptions{})
+
+		cmd := NewCmdDev()
+		cmd.SilenceUsage = true
+		cmd.Execute()
+
+		t.CheckDeepEqual(true, opts.Tail)
+		t.CheckDeepEqual(true, opts.Force)
+		t.CheckDeepEqual(true, opts.Force)
```
Comment (briandealwis) [inline cmd/skaffold/app/cmd/dev_test.go:177]: `opts.EnableRPC`?

File: cmd/skaffold/app/cmd/run_test.go (hunk 1 above)
Comment (dgageot) [reply cmd/skaffold/app/cmd/run_test.go:37]: Good catch!. I'll fix that

**Pull request merged**